assets/js/main.js

```diff
@@ -0,0 +1,256 @@
+  // all functions related to drops
+
+  // create a drop at the top of the screen with random x
+  var createDrop = function () {
+    var $drop = $('<div class="drop"></div>');
+    var x = Math.floor(Math.random() * GRID_WIDTH);
+    var y = 0;
+    move($drop, x, y);
+    size($drop, 1, 1);
+    $game.append($drop);
+    $drops.push($drop);
+    return $drop;
+  };
+
+  // remove the given drop
+  var removeDrop = function ($drop) {
+    $drops.splice($drops.indexOf($drop), 1);
+    $drop.remove();
+  };
+
+  // move all the drops
+  var moveDrops = function () {
+    for (var i = 0; i < $drops.length; i++) {
+      var $drop = $drops[i];
+      var pos = getXY($drop);
+      // move only by one to make collision detection easier
+      move($drop, pos.x, pos.y + 1);
+    }
+  };
+
+  // check if the drops are colliding with pad or bottom of screen
+  var colideDrops = function () {
+    var padPos = getXY($pad);
+    var $missed = [];
+
+    for (var i = 0; i < $drops.length; i++) {
+      var $drop = $drops[i];
+      var pos = getXY($drop);
+      // the drop hit the bottom :(
+      if (pos.y >= GRID_HEIGHT) {
+        $missed.push($drop);
+      } else if (
+        // colliding with the pad
+        pos.x >= padPos.x && pos.x <= padPos.x + game.padWidth &&
+        // out of bounds
+        pos.y === padPos.y) {
+        onDropCatch($drop);
+      }
+    }
+
+    // drop removeal is separated to avoid removing from an array midway
+    // through iterating
+    for (var i = 0; i < $missed.length; i++) {
+      var $drop = $missed[i];
+      onDropMiss($drop);
+    }
+  };
+
+  // ================== pad ==================
+  // pad related functions
+
+  var movePad = function () {
+    var pos = getXY($pad);
+    var newPos = pos.x + game.padDX;
+    // don't move the pad if it's out of bounds
+    if (newPos + game.padWidth <= GRID_WIDTH && newPos >= 0) {
+      move($pad, newPos, pos.y);
+    }
+  };
+
+  // ================== actions ==================
+  // game actions
+
+  var onDropMiss = function ($drop) {
+    removeDrop($drop);
+  };
+
+  var onDropCatch = function ($drop) {
+    playSound(catchSound);
+    game.score = game.score + 1;
+    $score.html(game.score);
+    removeDrop($drop);
+  };
+
+  // NOTE: we must use key down and up as keypress doesn't work for arrow keys
+  // set the pads movement direction based on key down
+  var onKeyDown = function (event) {
+    if (event.keyCode === LEFT) {
+      game.padDX = -1;
+    } else if (event.keyCode === RIGHT) {
+      game.padDX = 1;
+    }
+  };
+
+  // when the user lets go of the key set the pads movement to 0
+  var onKeyUp = function (event) {
+    // only reset the speed when keyup for the specific direction
+    // this handles when the player simultaneusly presses left and right
+    // also ignore other keyup events that are not left and right
+    if (event.keyCode === LEFT && game.padDX === -1 ||
+      event.keyCode === RIGHT && game.padDX === 1) {
```

Comment:
This is a more advanced bug fix.  The version without this works basically fine.